packages/did-comm/src/didcomm.ts

```diff
@@ -194,7 +194,19 @@ export class DIDComm implements IAgentPlugin {
     }
 
     // obtain sender signing key(s) from authentication section
-    const senderKeys = await mapIdentifierKeysToDoc(managedSender, 'authentication', context)
+
+    // can't sign did-comm message using web3 kms
+    let senderKeys: _ExtendedIKey[] = (await mapIdentifierKeysToDoc(managedSender, 'verificationMethod', context)) //.filter(k => k.kms !== 'web3')
```

Comment:
please also flag this as a workaround, as technically picking from `keyAgreement` is correct;
it's just very very picky, and annoying, but from a security perspective, it does reflect the intent of the DID controller for the purpose of keys.

Feel free to ignore this comment if this PR is only experimental

Reply:
Can you expand on this a bit? It would seem like there's a bug somewhere in did-ethr if `keyAgreement` is correct. Why would the Managed DID not include the new key in keyAgreement after I've added it? Am I adding the key incorrectly?

I'd like to find a more permanent solution, as this workaround causes some tests to fail

Reply:
I meant that using keys from their correct verification relationship should be the default.

It's usually [not safe to use the same key for multiple purposes](https://github.com/christianlundkvist/blog/blob/master/2020_05_26_secp256k1_twist_attacks/secp256k1_twist_attacks.md). If you add a secp256k1 key to the `assertionMethod` relationship, it shouldn't be also used for `keyAgreement`, for example.

Picking keys from the `verificationMethod` pool instead of from the `keyAgreement` or `authentication` sections respectively is a workaround.

We do have a bug in the `did-provider-ethr` in that we don't allow the user to decide where to add new keys, and just assume based on the key type.

To simplify some of the work needed to get did-comm working using two did:ethr, I also created this issue: https://github.com/decentralized-identity/ethr-did-resolver/issues/174